test/e2e/framework/nodes.go

```diff
@@ -68,3 +69,37 @@ func (f *Framework) SetGatewayLabelOnNode(cluster ClusterIndex, nodeName string,
 			return err
 		})
 }
+
+// FindAnyNonGatewayRouteAgentPodNodes looks for the route agent pod on any cluster node, including control plane
+// The function will be used in upstream ci environment, lack of resources where two GW nodes are used.
+func (f *Framework) FindAnyNonGatewayRouteAgentPodNodes(cluster ClusterIndex) []*v1.Node {
+	gwNodesSet := stringset.New()
```

Comment:
You can simply use a map here instead of a set. 
 
```suggestion
	gwNodesSet := make(map[string]bool)
```

Reply:
You're right.
Thanks for the idea.

Reply:
Replaced.